### PR TITLE
[RTR] split final MHE save

### PR DESCRIPTION
**Pull request by Ipuch**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document [docs/contribution.md]?
* [x] Have you checked to ensure there aren't other open [Pull Requests] for the same update/change?
* [x] Have you opened/linked the issue related to your pull request?
* [x] Have you used the tag [WIP] for on-going changes, and removed it when the pull request was ready?
* [x] When ready to merge, have you sent a comment pinging @pariterre in it?

### New Feature Submissions:

1. [x] Does your submission pass the tests (if not please explain why this is intended)?
2. [x] Did you write a proper documentation (docstrings and ReadMe)
3. [x] Have you linted your code locally prior to submission (using the command: `black . -l120 --exclude "external/*"`)?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new examples for your core changes, as applicable?
* [x] Have you written new tests for your core changes, as applicable?

- - -
This change is [Reviewable](https://reviewable.io/reviews/pyomeca/bioptim/592)

### CODE REVIEW

**Comment by Ipuch** [comment]
won't work yet because, when we move from one frame to another it make no sens to same this.
Only if frame to move is superior to 1, or only for multicyclic.

File: bioptim/optimization/receding_horizon_optimization.py [hunk 1]
```diff
@@ -512,6 +515,40 @@ def advance_window_initial_guess_controls(self, sol, **advance_options):
...
+            update_function=update_function, **extra_options
+        )
+
+        if get_cycles:
+            cycle_solutions = []
+            for sol in solution[1]:
+                _states, _controls = self.export_cycles(sol)
+                cycle_solutions.append(self._initialize_one_cycle(_states, _controls))
+
+        if get_all_iterations and get_cycles:
```
Comment (pariterre) [inline bioptim/optimization/receding_horizon_optimization.py:538]: out = []
if get_all_interations:
    out.append(...)
if ...:
    out.append(...)

Comment (Ipuch) [reply bioptim/optimization/receding_horizon_optimization.py]: Done.

**Comment by Ipuch** [comment]
@pariterre

**Review by pariterre — APPROVED** [review]

**Pull request merged**